File: Lab Final/TaskManager/src/IssueRepository.java
import java.util.HashMap;
import java.util.Map;

public class IssueRepository {
    private Map<Integer, Issue> issues;

    public IssueRepository() {
        issues = new HashMap<>();
    }

    public void addIssue(Issue issue) {
        issues.put(issue.ID, issue);
    }

    public Issue getIssue(int id) {
        return issues.get(id);
    }

    public void removeIssue(int id) {
        issues.remove(id);
    }

    public Map<Integer, Issue> getAllIssues() {
        return issues;
    }
}

File: TaskManager/MyTaskManager/IssueRepository.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MyTaskManager
{
    public class IssueRepository
    {
        public Dictionary<int, Issue> issues;
        public int nextIssueId;
        public int nextCommentId;

        public IssueRepository()
        {
            issues = new Dictionary<int, Issue>();
            nextIssueId = 1;
            nextCommentId = 1;
        }

        public int GetNextIssueId()
        {
            return nextIssueId++;
        }

        public int GetNextCommentId()
        {
            return nextCommentId++;
        }

        public void AddIssue(Issue issue)
        {
            issues[issue.Id] = issue;
        }

        public Issue GetIssue(int id)
        {
            if (issues.ContainsKey(id)) return issues[id];
            return null;
        }

        public void UpdateIssue(Issue issue)
        {
            if (issues.ContainsKey(issue.Id))
            {
                issues[issue.Id] = issue;
            }
        }

        public Dictionary<int, Issue> GetAllIssues()
        {
            return issues;
        }
    }
}
